### ISSUE TITLE
type error of versionCode

### ISSUE THREAD

**Issue by screw-hand** [uni-helper/uni-app-schemas-vscode#15]
### Describe the bug

`versionCode` should be a `number`, not a `string` type.

when I input `'versionCode': '190'` in `manifest.config.ts`，It good for `vite-plugin-uni-manifest`.
But I got a warn in `src/manifest.json` => `Incorrect type. Expected "number"` in vscode.

**the type warn is only show in vscode, HBuilderX everything ok.**

<img width="1011" alt="image" src="https://github.com/uni-helper/vite-plugin-uni-manifest/assets/26453811/a1b4068d-1656-4b3f-a268-c4704aaa674d">

![image](https://github.com/uni-helper/vite-plugin-uni-manifest/assets/26453811/0d2ea2af-575f-4cc8-a514-97130597b0d2)

### Reproduction

# manifest.config.ts

```ts
import { defineManifestConfig } from '@uni-helper/vite-plugin-uni-manifest'

export default defineManifestConfig({
  'name': 'vitesse-uni-app',
  'appid': '',
  'description': '',
  'versionName': '1.9.0',
  'versionCode': '190',
  'transformPx': false,
  'app-plus': {
  },
  'vueVersion': '3',
})
```

# src/manifest.json

```
{
  "name": "vitesse-uni-app",
  "appid": "",
  "description": "",
  "versionName": "1.9.0",
  "versionCode": "190",
  "transformPx": false,
  "app-plus": {},
  "vueVersion": "3"
}
```

### System Info

```Shell
# npx envinfo --system --binaries --browsers
  System:
    OS: macOS 13.2.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 39.36 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.20.0 - ~/.nvm/versions/node/v16.20.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.20.0/bin/yarn
    npm: 9.6.6 - ~/.nvm/versions/node/v16.20.0/bin/npm
    Watchman: 2023.03.06.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 111.0.5563.146
    Edge: 113.0.1774.42
    Safari: 16.3

#./node_modules/.bin/tsc -v
Version 5.0.4

# HBuilder X version
3.7.11.20230427

# vscode --version
1.78.2
b3e4e68a0bc097f0ae7907b217c1119af9e03435
arm64
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/uni-helper/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by KeJunMao** [comment]
cc: @ModyQyW

**Comment by screw-hand** [comment]
I have identified the location of the problem. 

Just change the code in `node_modules/@uni-helper/vite-plugin-uni-manifest/dist/index.d.ts`

```typescript
=== before ==
/** 版本号 */
versionCode: string;

=== after ===
/** 版本号 now is number */
versionCode: number;
```

<img width="1249" alt="image" src="https://github.com/uni-helper/uni-app-schemas-vscode/assets/26453811/cc42aa69-69a3-4a47-b06e-1e0908f2c55f">

**Comment by screw-hand** [comment]
https://github.com/uni-helper/vite-plugin-uni-manifest/blob/e736a899b5fe60768b64828940b8b9e2303b30c1/src/config/types.ts#L672

just change the `string` to `number` can fix this warning.

@KeJunMao

**Comment by KeJunMao** [comment]
thanks, but `versionCode` is should allow strings

see official [full-manifest](https://uniapp.dcloud.net.cn/collocation/manifest-app.html#full-manifest)

**Comment by KeJunMao** [comment]
the better way is： 

https://github.com/uni-helper/uni-app-schemas-vscode/blob/f78850eed04cf2bbbaed239d2e218c9b6522c321/schemas/manifest.json#L28

```diff
{
- "type":  "number",
+ "type": ["string", "number"],
}
```

and change uni-helper/vite-plugin-uni-manifest `versionCode` allow strings

**Comment by screw-hand** [comment]
Great, thank you! I will continue to monitor this issue.

**Comment by ModyQyW** [comment]
Included in v0.8.11. Thanks @KeJunMao @screw-hand !

I have lots of things to do so I am a bit late. Sorry about that.
